lib/linkerd2-proxy-core/src/resolve.rs

```diff
@@ -1,31 +1,87 @@
 use futures::{Future, Poll};
+use linkerd2_error::Error;
 use std::net::SocketAddr;
 
 /// Resolves `T`-typed names/addresses as a `Resolution`.
 pub trait Resolve<T> {
```

Comment:
This trait alias exists to expose the Endpoint type so that type assertions are less crazy. Any service that returns a resolution automatically implements this. But, this permits
```rust
    R: Resolve<DstAddr, Endpoint = Endpoint>,
```
without this it would be
```rust
    R: Service<DstAddr>,
    R::Response: Resolution<Endpoint = Endpoint>
,
```

I find it especially useful to make aliases for special kinds of services, because otherwise every type bounds is just a huge mess of `Service` and it gets very confusing very fast. Though trait aliases aren't a panacea, being able to bundle a bunch of type assertions under a reusable handle is extremely helpful.